### DISCUSSION TITLE
How to get the YOLO badge

### DISCUSSION THREAD

**Discussion by leozz37** [Schweinepriester/github-profile-achievements#6] [Q&A] [answered]
Any idea on how to unlock the `YOLO` badge? I tried merging a PR without reviewing it as the badge description says but didn't work.

EDIT (23/06/22): seems like Github have loosened the criteria, a lot of people it's getting just for opening a PR and merging it.

**Comment by mMarzeta** [comment]
I believe you need to set repo policy to always have approvals before merging. And then use admin "superpowers" to merge it anyway. This is how I've earned it.

**Reply by leozz37** [reply]
I tried and it didn't work :(

**Reply by csthenry** [reply]
Hi all,
for get starstruck need to reach 10 star. If u want u can put star on my readme repository, I will do the same for you.
https://github.com/csthenry/MagicLitePlanet

**Reply by WixkedLuck** [reply]
Thanks for the help!

**Reply by anekemichael** [reply]
Thanks so much.. It worked for me

**Reply by SammyLeths** [reply]
done @csthenry

**Reply by b0bd3v** [reply]
@csthenry done
https://github.com/b0bd3v/kafka_vs_rabbitmq

**Reply by b0bd3v** [reply]
@csthenry done
https://github.com/b0bd3v/kafka_vs_rabbitmq

**Comment by Schweinepriester** [comment]
My understanding is that somebody needs to be assigned as a reviewer to the PR (top right on desktop) and the PR then be merged without that person (or anybody else) doing a review.

**Reply by leozz37** [reply]
Awesome, I'll try!
If it works I let you know!

**Reply by leozz37** [reply]
Didn't work :(

**Reply by Schweinepriester** [reply]
Were you the author of the PR and not also yourself the reviewer?
Maybe even additionally didn't assign yourself?

**Reply by leozz37** [reply]
Three people were involved.

- One opened the PR and requested a review from the second;
- The second was requested to review, but did not review
- The third (me) merged the PR

**Reply by Schweinepriester** [reply]
I would have thought that you would have to be person one and maybe also person three.

But yeah, as you note down in another reply it seems kinda flaky currently…

**Reply by leozz37** [reply]
Tried as well and didn't work 😔

**Reply by rickstaa** [reply]
I achieved this by merging a pull request, without a review, on a repository (i.e. https://github.com/anuraghazra/github-readme-stats/pull/1605), of which I'm not the owner but merely a contributor.

![image](https://user-images.githubusercontent.com/17570430/173228272-faa73207-2cb6-45d5-8ab7-bc27b99d1c53.png)

I hope that helps you unfold the requirements for achieving this badge.

**Reply by csthenry** [reply]
Hi all,
for get starstruck need to reach 10 star. If u want u can put star on my readme repository, I will do the same for you.
https://github.com/csthenry/MagicLitePlanet

**Reply by vaibhavx42** [reply]
done @csthenry

**Reply by Presstomsk** [reply]
Thank U

**Comment by ajmeese7** [comment]
I got the badge without doing any of the things mentioned above, I simply made commits on another branch to add a feature and merged the branch into main. You can see my actual PR [here](https://github.com/ajmeese7/8bit/pull/1), as evidenced by the badge information on my user profile.

Accept if this helps, I'm looking forwards to getting the discussion award too 😄

**Reply by Schweinepriester** [reply]
but whats different to https://github.com/Schweinepriester/github-profile-achievements/pull/4? 🤔 

as I'm typing I'm thinking: maybe the way it's merged is crucial as well - because I merged with "rebase and merge" using the button on github. do you remember? did you merge manually on the CLI or didn't use rebase or something?

**Reply by qiqt** [reply]
"squash and merge" worked for me

**Reply by leozz37** [reply]
I think it's kinda "random"... Some people got the badge just merging a random repo and some others got the merged PR badge instead...

**Reply by Sleggi** [reply]
Perhaps it really is some kind of random, or you had to take some steps before

**Reply by pingshunhuangalex** [reply]
Tried all the suggestions above, but still no luck. It's driving me crazy...

**Reply by rickstaa** [reply]
@pingshunhuangalex Did you try the steps in https://github.com/Schweinepriester/github-profile-achievements/discussions/6#discussioncomment-2932109? 

- Creating PR on a repository in which you are not the owner but a collaborator.
- Request review from the owner.
- Then, merge it without waiting for the review.

I got it by doing these steps. I think it makes sense with the name YOLO since it is against most generally accepted contribution guidelines, but in https://github.com/anuraghazra/github-readme-stats/pull/1605 I did it anyway since the PR was minimal, and I was tired of waiting for a review. 😈

**Reply by pingshunhuangalex** [reply]
Got it now. Thanks @rickstaa 

The key here is having someone as the reviewer. Without the reviewer, the PR won't go into the code review state which is actually displayed in the GitHub UI.

Thanks.

**Reply by csthenry** [reply]
Hi all,
for get starstruck need to reach 10 star. If u want u can put star on my readme repository, I will do the same for you.
https://github.com/csthenry/MagicLitePlanet

**Comment by ghsable** [comment]
- Pull request ( Checks 0 )
- Merge Pull request ( Create merge commit )

**Reply by csthenry** [reply]
Hi all,
for get starstruck need to reach 10 star. If u want u can put star on my readme repository, I will do the same for you.
https://github.com/csthenry/MagicLitePlanet

**Comment by augustobr1to** [comment]
Create a pull request and merge.

**Comment by Schweinepriester** [comment, marked as answer]
2022-06-23, EDIT:
1. Create a PR on your own repository
2. Merge it
3. Done!

Thanks to @Maxim-Mazurok & @supermavster for testing and @gabodmc for the actual [steps I stole from below](https://github.com/Schweinepriester/github-profile-achievements/discussions/6#discussioncomment-3010096).

They seem to have loosened the criteria to not require a reviewer to be assigned anymore.

---

Alright, thanks to @ale-lit we have two more examples and I think the criteria.
Examples:
- https://github.com/Schweinepriester/achievement-testarea/pull/4 for @Schweinepriester 
- https://github.com/ale-lit/book-shop-laravel/pull/4 for @ale-lit

The only criteria seems to be to have someone else assigned as the reviewer (isn't possible to assign yourself anyway) and then merge without a review.

Additional information deduced from those PRs:
- it can be your own repo
- it can be only one line changed
- default commit message, default PR title and empty description is possible
- the achievement is awarded instantly
- I merged using the usual "Merge pull request" with default option "Create a merge commit" selected (not sure about the other options)
- For my PR: At the time of the merge, the reviewer was a collaborator of the repo (previously invited and accepted, so I could request a review)
- My repo had default settings and was created on 2022-06-10 (in case defaults change) - mentioning it because of the [repo policy answer](https://github.com/Schweinepriester/github-profile-achievements/discussions/6#discussioncomment-2923143)

I'm gonna mark this as the answer, at least for now.
If it doesn't work with those criteria, please link the PR if possible.

**Reply by leozz37** [reply]
Worked for me! My [PR](https://github.com/leozz37/jaguar/pull/5) for evidence!

- I own de repo
- Repo with at least one approve policy
- Opened a one line change PR with default description
- Waited for my pipelines to finish
- Requested a review from a collaborator (invited) of my repo
- Merged using the usual "Merge pull request" option

**Reply by Maxim-Mazurok** [reply]
Not sure then I got it, but github says that I got it for merging this: https://github.com/Maxim-Mazurok/eslint-plugin-vue-2-object-reactivity/pull/2 and it never had reviewers assigned, so you might want to edit this part "to have someone else assigned as the reviewer"

**Reply by Schweinepriester** [reply]
Thank you for the feedback, it really seems they have loosened the criteria!

Another recent example without any reviewer: https://github.com/supermavster/achievement-testarea/pull/1

But I had tried before without reviewer and I didn't work, for example: https://github.com/Schweinepriester/achievement-testarea/pull/1

I will edit the answer.

**Reply by csthenry** [reply]
Hi all,
for get starstruck need to reach 10 star. If u want u can put star on my readme repository, I will do the same for you.
https://github.com/csthenry/MagicLitePlanet

**Reply by guidorapisarda** [reply]
Nice!

**Reply by RamkrishnaSawant** [reply]
Hello,
    I am completing my starstruck achievement so put the star on my readme repository. I will do same for you
    https://github.com/RamkrishnaSawant

**Reply by gindevp** [reply]
nice

**Reply by rodrigogregorioneri** [reply]
help me get achievements by donating stars that i return with star:

https://github.com/rodrigogregorioneri/serverless_integration

**Reply by cxman1** [reply]
I'm getting the achievement badge of the star chasing clan. Please add an asterisk to me and I will add it back to you
https://github.com/cxman1/vue-collect

**Reply by cxman1** [reply]
> 通过捐赠我与明星一起返回的明星来帮助我获得成就：
> 
> https://github.com/rodrigogregorioneri/serverless_integration

I've added your warehouse

**Reply by cxman1** [reply]
> 大家好， 要追星需要达到10星。如果你想要你可以在我的自述文件库中添加星号，我也会为你做同样的事情。 https://github.com/csthenry/MagicLitePlanet

I have added your warehouse

**Reply by rodrigogregorioneri** [reply]
@cxman1 done

**Reply by nchdatta** [reply]
Help me get achievements by donating stars that I'll return with star: https://github.com/nchdatta/personal-portfolio

**Reply by RatulHasan** [reply]
### That is cool to know! Thank you! :)

**Reply by fgmn** [reply]
I star yours, here is mine: https://github.com/fgmn/my-sdu

**Comment by nshen** [comment]
I got my YOLO badge
1. Invite another account to your repo by Settings -> Collaborators -> Add people 
2. Create a PR, and add a Reviewer by selecting the user you invited.
3. merge PR without review.

**Reply by genediazjr** [reply]
does it have to be a public repo?

**Reply by VitRod** [reply]
On the  3rd step I have merged PR without review and  it has worked. I got a  badge.

**Reply by ghost** [reply]
Thanks for the info.

**Reply by xxtaichi** [reply]
It's great!

**Reply by winsphinx** [reply]
Thanks!

**Reply by sibeux** [reply]
work, easy to understand, thankyou very much

**Reply by ddh** [reply]
Worked for me

**Reply by faintout** [reply]
thanks

**Reply by rodrigogregorioneri** [reply]
help me get achievements by donating stars that i return with star:

https://github.com/rodrigogregorioneri/serverless_integration

**Reply by hctian713** [reply]
thankS!!!!

**Reply by adrianschubek** [reply]
thx

**Reply by nchdatta** [reply]
Help me get achievements by donating stars that I'll return with star: https://github.com/nchdatta/personal-portfolio

**Comment by lincolnvc** [comment]
You need to submit two pull requests for review and merge them later

**Reply by csthenry** [reply]
Hi all,
for get starstruck need to reach 10 star. If u want u can put star on my readme repository, I will do the same for you.
https://github.com/csthenry/MagicLitePlanet

**Reply by Romic77** [reply]
Hi all,
for get starstruck need to reach 10 star. If u want u can put star on my readme repository, I will do the same for you.
https://github.com/Romic77/go-blog

**Comment by gabodmc** [comment]
1) Create a PR on your own repository
2) Merge them
3) Done!

**Reply by ankushshinde755** [reply]
worked for me !!

**Comment by smir45** [comment]
yeah creating pull request on our own repository and merging them actually works

**Comment by GoHoo** [comment]
Dear god can someone make a quick video/image tutorial or something, because I cant get this for the life of me

**Reply by quartesa37** [reply]
> Dear god can someone make a quick video/image tutorial or something, because I cant get this for the life of me

**Reply by csthenry** [reply]
Hi all,
for get starstruck need to reach 10 star. If u want u can put star on my readme repository, I will do the same for you.
https://github.com/csthenry/MagicLitePlanet

**Reply by arijhajji-1** [reply]
hi @csthenry  i can put 10 stars on your readme can u do the same for me?

**Comment by seryy366** [comment]
test

**Comment by Hauzenex** [comment]
i got it thanks es

**Comment by sourabmaity** [comment]
test

**Reply by csthenry** [reply]
Hi all,
for get starstruck need to reach 10 star. If u want u can put star on my readme repository, I will do the same for you.
https://github.com/csthenry/MagicLitePlanet

**Comment by cristicretu** [comment]
Another thing you can do, without collaborators is to create a [pull request template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository) and then do the steps (create a pr -> merge it).

This is what I did on my repository and it worked, without me even knowing it 😅

**Comment by SunsetMkt** [comment]
I created a branch in my own repo and created a pull request to myself and merged it.

**Comment by quartesa37** [comment]
> Any idea on how to unlock the `YOLO` badge? I tried merging a PR without reviewing it as the badge description says but didn't work.
> 
> EDIT (23/06/22): seems like Github have loosened the criteria, a lot of people it's getting just for opening a PR and merging it.

**Reply by csthenry** [reply]
Hi all,
for get starstruck need to reach 10 star. If u want u can put star on my readme repository, I will do the same for you.
https://github.com/csthenry/MagicLitePlanet

**Comment by ashishantonytsr** [comment]
Hi I am just testing 
Mark my answer please

**Comment by newlomar** [comment]
For the ones reading this now, I would like to add a information that I did not see anyone mention before that made me struggle:

you have to add a reviewer, so you need to have a repository with more than one contributor.

**Comment by G-Redos** [comment]
Hi, testing.

**Comment by kensoz** [comment]
I created a branch in my own repo and created a pull request to myself and merged it.
but it's not work

**Comment by vaibhavx42** [comment]
Did you added another account as collaborator of not try this
On your own repository go to settings add another account as collaborator
Now on  your own repository make a pr and add the another collaborator
account as reviewer
Then merge that pr from this account only without reviewing.

**Reply by kensoz** [reply]
thanks bro. I will try it

**Comment by Fanor51** [comment]
This does not work if you got a github CI Linter or something else running.

**Comment by Kushal997-das** [comment]
Hello I got the answer and it's perfectly working - 

STEP 1 :  Choose a normal repo. of yours remember in that repo there shouldn't be any `.github` folder or any auto running things not present .

STEP 2 : Now go to that repository settings add another account as collaborator . (Settings->collaborator->Add people).

STEP 3 :  You have to Create a Pull request (PR)  on this repository. 

STEP 4 : Add that  collaborator account as reviewer.

STEP 5 : Then You have to merge your opned PR  from Your account only not collaborator , Without reviewing the PR .

I hope it gonna help you guys , Thanks :)

**Reply by myfix16** [reply]
Jan. 14, 2023, it works. It seems that now a reviewer is required for this achievement.

**Reply by AnthonyNet** [reply]
Finally, thank you :-)

**Reply by kylekce** [reply]
It works! The only different thing is I added more than one collaborator. But it MIGHT still work with only one collaborator.

**Reply by hamzakhan663** [reply]
Works!

**Comment by m0-dev** [comment]
Hi, i'm testing.
